system/lib/pthread/pthread_testcancel.c

```diff
@@ -8,15 +8,11 @@
 #include "pthread_impl.h"
 #include <pthread.h>
 
-int _pthread_isduecanceled(struct pthread* pthread_ptr) {
-  return pthread_ptr->cancel != 0;
-}
-
 void __pthread_testcancel() {
   struct pthread* self = pthread_self();
   if (self->canceldisable)
     return;
-  if (_pthread_isduecanceled(self)) {
```

Comment:
Can the removal of _pthread_isduecanceled also be split out into its own PR?

Reply:
Reverted with commit 8db22b9e14153eae72d20df98aa6d2673807a8f5, moved to PR #15625.